### ISSUE TITLE
[Elastic Agent] Add support for Fleet-Server running on a Mac with an M1 chip

### ISSUE THREAD

**Issue by nimarezainia** [elastic/beats#26684]
**Describe the enhancement:**
As part of a greater effort to have beats and agent to build natively on the new M1 chip set.

**Describe a specific use case for the enhancement or feature:**
As a user I may need to deploy the fleet-server on a dedicated Mac machine

**Comment by AndersonQ** [comment]
@nimarezainia wouldn't be better to move this issue to [fleet-server's](https://github.com/elastic/fleet-server) repo?

**Comment by jlind23** [comment]
@AndersonQ can I close this one following the work you did on the M1?

**Comment by AndersonQ** [comment]
no, fleet-server is still missing. We talked that we could merge the beats and elastic-agent PRs first as it's unlikely someone will run fleet-server on a M1 (at least in a production environment)
